src/parsers/manifest/dash/node_parsers/__tests__/AdaptationSet.test.ts

```diff
@@ -0,0 +1,330 @@
+    const element1 = new DOMParser()
+      .parseFromString(`<AdaptationSet ${attributeName}="012" />`, "text/xml")
+      .childNodes[0] as Element;
+    expect(createAdaptationSetIntermediateRepresentation(element1))
+      .toEqual({
+        attributes: { [_variableName]: 12 },
+        children: { baseURL: "", representations: [] },
+      });
+
+    const element2 = new DOMParser()
+      .parseFromString(`<AdaptationSet ${attributeName}="0" />`, "text/xml")
+      .childNodes[0] as Element;
+    expect(createAdaptationSetIntermediateRepresentation(element2))
+      .toEqual({
+        attributes: { [_variableName]: 0 },
+        children: { baseURL: "", representations: [] },
+      });
+    const element3 = new DOMParser()
+      .parseFromString(`<AdaptationSet ${attributeName}="-50" />`, "text/xml")
+      .childNodes[0] as Element;
+    expect(createAdaptationSetIntermediateRepresentation(element3))
+      .toEqual({
+        attributes: { [_variableName]: -50 },
+        children: { baseURL: "", representations: [] },
+      });
+    expect(spyLog).not.toHaveBeenCalled();
+    spyLog.mockRestore();
+  });
+
+  /* tslint:disable max-line-length */
+  it(`should correctly parse an AdaptationSet element with an incorrect ${attributeName} attribute`, () => {
+  /* tslint:enable max-line-length */
+    const spyLog = jest.spyOn(log, "warn").mockImplementation(jest.fn());
+    const element1 = new DOMParser()
+      .parseFromString(`<AdaptationSet ${attributeName}="toto" />`, "text/xml")
+      .childNodes[0] as Element;
+    expect(createAdaptationSetIntermediateRepresentation(element1))
+      .toEqual({
+        attributes: {},
+        children: { baseURL: "", representations: [] },
+      });
+    expect(spyLog).toHaveBeenCalledTimes(1);
+    expect(spyLog).toHaveBeenCalledWith(`DASH: invalid ${attributeName} ("toto")`);
+
+    const element2 = new DOMParser()
+      .parseFromString(`<AdaptationSet ${attributeName}="PT5M" />`, "text/xml")
+      .childNodes[0] as Element;
+    expect(createAdaptationSetIntermediateRepresentation(element2))
+      .toEqual({
+        attributes: {},
+        children: { baseURL: "", representations: [] },
+      });
+    expect(spyLog).toHaveBeenCalledTimes(2);
+    expect(spyLog).toHaveBeenCalledWith(`DASH: invalid ${attributeName} ("PT5M")`);
+
+    const element3 = new DOMParser()
+      .parseFromString(`<AdaptationSet ${attributeName}="" />`, "text/xml")
+      .childNodes[0] as Element;
+
+    expect(createAdaptationSetIntermediateRepresentation(element3))
+      .toEqual({
+        attributes: {},
+        children: { baseURL: "", representations: [] },
+      });
+    expect(spyLog).toHaveBeenCalledTimes(3);
+    expect(spyLog).toHaveBeenCalledWith(`DASH: invalid ${attributeName} ("")`);
+    spyLog.mockRestore();
+  });
+
+  /* tslint:disable max-line-length */
+  it(`should correctly parse an AdaptationSet element with a boolean ${attributeName} attribute`, () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 26. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=445&issues=AWgzizTrSi5GWQS9TjAW&open=AWgzizTrSi5GWQS9TjAW)

[ISSUE_KEY]: AWgzizTrSi5GWQS9TjAW